fraud/service.go

```diff
@@ -0,0 +1,171 @@
+package fraud
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/celestiaorg/celestia-node/header"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+var log = logging.Logger("fraud")
+
+// proofUnmarshaler aliases a function that parses data to `Proof`
+type proofUnmarshaler func([]byte) (Proof, error)
+
+// headerFetcher aliases a function that is used to fetch ExtendedHeader from store
+type headerFetcher func(context.Context, uint64) (*header.ExtendedHeader, error)
+
+// Validator aliases a function that validate pubsub incoming messages
+type Validator func(ctx context.Context, proofType ProofType, data []byte) pubsub.ValidationResult
```

Comment:
Same here. Lets' move it to the interface.go

Reply:
Moved to interface.go